### ISSUE TITLE
Avoid bad paths when walking

### ISSUE THREAD

**Issue by rindlerblabla** [osmandapp/Osmand#4032]
Would be nice with an option when using pedestrian routing to avoid paths with smoothness bad/terrible and also trail_visibility bad/terrible.

**Comment by jfburke3** [comment]
Unlit paths might also be good to avoid at night.

As an addition, it might be useful to start looking into how to integrate opening_hours for routes (or at least barriers).  What I'm thinking of when I say this is I know of some places where the shortest foot route is through a national forest area on some well-maintained trails.  However, they are legally closed from dusk to dawn, so you wouldn't want to route someone through the area at night.

**Comment by andrasfuchs** [comment]
This could be useful for the cycleways as well. It the smoothness is bad, it could be indicated while planning the route. 

Also, using the sensors in the phone it could also record the smoothness of the cycleway if the users allowed the upload of sensory data collected during his/her trip.

**Comment by ChrisLane** [comment]
Certainly, being able to calculate a walking route avoiding unlit paths would be great.

**Comment by scaidermern** [comment]
Adding the possibility to avoid ways with bad smoothness will also be useful for implementing #9415.
